src/wazuh_modules/vulnerability_detector/wm_vuln_detector_nvd.c

```diff
@@ -2335,17 +2330,15 @@ int wm_vuldet_check_generic_package(sqlite3 *dbCVE, vu_feed feed, char *pkg_name
     bool app_package_mac = false;
 
     pkg_reference = (name_type == PACKAGE_SOURCE) ? pkg_source : pkg_name;
+    int product_name_type = wm_vuldet_is_product_name(pkg_reference);
 
-    if (!strcmp(pkg_reference, PR_UBUNTU) ||
-        !strcmp(pkg_reference, PR_DEBIAN) ||
-        !strcmp(pkg_reference, PR_REDHAT) ||
-        !strcmp(pkg_reference, PR_ARCH)   ||
-        !strcmp(pkg_reference, PR_KERNEL)) {
+    if (product_name_type != PR_NOSYSTEM) {
         query = vu_queries[VU_GET_GENERIC_PACKAGE_OS];
         os_package = true;
     } else if (!strcmp(pkg_reference, "mac_os_x") || !strcmp(pkg_reference, "mac_os_x_server") ||
                !strcmp(pkg_reference, "mac_os") || !strcmp(pkg_reference, "macos")) {
         query = vu_queries[VU_GET_GENERIC_PACKAGE_OS_MAC];
+        os_package = true;
```

Comment:
This flag is to discard the OS vulnerabilities so it should be disabled for macOS agents